lib/spack/spack/modules/common.py

```diff
@@ -535,6 +535,7 @@ def long_description(self):
     @tengine.context_property
     def configure_options(self):
         pkg = self.spec.package
+        build_environment.setup_package(pkg, False)
```

Comment:
I don't think we should call this function here, as it will modify the current environment:

https://github.com/spack/spack/blob/553acbbe3dfc797f5c9fc3fabd701d016c63ba33/lib/spack/spack/build_environment.py#L723

During a normal installation this is done in a forked process:

https://github.com/spack/spack/blob/553acbbe3dfc797f5c9fc3fabd701d016c63ba33/lib/spack/spack/package.py#L1477-L1546

so the modifications will disappear when going to the next package. When we are refreshing module files I don't think this will be the case, so modifications will stack-up in a sort of random manner.

The obvious solution to this would be to also fork when writing module files, but I think we might take the chance to go back to the drawing board and see if we can find a better solution than these hotfixes.

Reply:
@michaelkuhn I thought about this a bit, and the best solution that comes to my mind is to log the configure line somewhere in `<prefix>/.spack`. This would have at least two benefits:
1. We don't have to recreate the build-environment and spawn a different process just to get the configuration line again
2. We'll record the exact configuration line used during the build, and will be immune to spurious changes due to e.g. a different environment when refreshing the module wrt when building the package in the first place

What do you think about this solution? Do you see any issue with that?

Reply:
> What do you think about this solution? Do you see any issue with that?

Not at all, I actually had a similar idea when I created the PR but assumed there was a good reason for including it in the module file.

Now that you mention it, `build.out` actually already contains the configure call with arguments. Do you think this is enough?

I can update the PR later to remove the `configure_options` stuff if you want.

Reply:
> Now that you mention it, build.out actually already contains the configure call with arguments. Do you think this is enough?

`build.out` contains the configure call, but that needs to be identified and parsed - it might be simple, but we also might encounter pathological builds that require adjusting regexes or that have large logs to parse, etc. 

I would modify the code that dumps the provenance to write the `configure_args` string somewhere, and then re-read it when needed. @scheibelp @tgamblin Any opinion on this?

Reply:
Ah, okay, got it! So we will keep the arguments in the module files but instead of regenerating them on demand, we just re-read them from a file.